src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/Language/Legacy/ParserTestBase.cs

```diff
@@ -306,116 +273,59 @@ internal RazorSyntaxTree ParseCodeBlock(string document, bool designTime = false
         {
-            ParseBlockTest(RazorLanguageVersion.Latest, document, directives, designTime, expectedErrors);
+            ParseBlockTest(RazorLanguageVersion.Latest, document, directives, designTime);
         }
 
-        internal virtual void ParseBlockTest(RazorLanguageVersion version, string document, IEnumerable<DirectiveDescriptor> directives, bool designTime, params RazorDiagnostic[] expectedErrors)
+        internal virtual void ParseBlockTest(RazorLanguageVersion version, string document, IEnumerable<DirectiveDescriptor> directives, bool designTime)
         {
             var result = ParseBlock(version, document, directives, designTime);
 
-            BaselineTest(result);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document)
-        {
-            SingleSpanBlockTest(document, default, default);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document, BlockKindInternal blockKind, SpanKindInternal spanType, AcceptedCharactersInternal acceptedCharacters = AcceptedCharactersInternal.Any)
-        {
-            SingleSpanBlockTest(document, blockKind, spanType, acceptedCharacters, expectedError: null);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document, string spanContent, BlockKindInternal blockKind, SpanKindInternal spanType, AcceptedCharactersInternal acceptedCharacters = AcceptedCharactersInternal.Any)
-        {
-            SingleSpanBlockTest(document, spanContent, blockKind, spanType, acceptedCharacters, expectedErrors: null);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document, BlockKindInternal blockKind, SpanKindInternal spanType, params RazorDiagnostic[] expectedError)
-        {
-            SingleSpanBlockTest(document, document, blockKind, spanType, expectedError);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document, string spanContent, BlockKindInternal blockKind, SpanKindInternal spanType, params RazorDiagnostic[] expectedErrors)
-        {
-            SingleSpanBlockTest(document, spanContent, blockKind, spanType, AcceptedCharactersInternal.Any, expectedErrors ?? new RazorDiagnostic[0]);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document, BlockKindInternal blockKind, SpanKindInternal spanType, AcceptedCharactersInternal acceptedCharacters, params RazorDiagnostic[] expectedError)
-        {
-            SingleSpanBlockTest(document, document, blockKind, spanType, acceptedCharacters, expectedError);
-        }
-
-        internal virtual void SingleSpanBlockTest(string document, string spanContent, BlockKindInternal blockKind, SpanKindInternal spanType, AcceptedCharactersInternal acceptedCharacters, params RazorDiagnostic[] expectedErrors)
-        {
-            var result = ParseBlock(document, designTime: false);
-
-            BaselineTest(result);
+            BaselineTest(result, ensureFullFidelity: false);
```

Comment:
Ah, I see. So do we have the rollout of this tracked somewhere? I'd like to understand more.

Reply:
https://github.com/aspnet/AspNetCore/issues/6470 is still the issue for tracking this. I'm not going to close it with this PR.

>  I'd like to understand more.

I assume you meant the reason for failures. [Here](https://github.com/aspnet/AspNetCore-Tooling/blob/master/src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/HtmlTagsTest.cs#L78) is an example of a failing test,
```C#
ParseBlockTest("<![CDATA[Foo]]> Bar");
//              ^              ^
```
In this case, `ParseBlock` is satisfied once it finishes parsing the CDATA tag and returns. `Bar` is never parsed and hence fails the full fidelity check.

Reply:
OK Cool. Alternate suggestion for aspnet/AspNetCore#6470 then. 

For the `ParseBlock` tests change the check to a `StartsWith`, and add to the test setup to validate the trailing part of the document as well. This doesn't require us to fundamentally alter what's being tested, but it will make the tests really robust.

Reply:
Ooh I like that 👍